src/parser/Parser.ts

```diff
@@ -877,49 +886,44 @@ export class Parser {
 
             this.consumeStatementSeparators(true);
 
-            let func = new FunctionExpression({
-                parameters: params,
-                body: undefined, //body
-                functionType: functionType,
-                endFunctionType: undefined, //ending keyword
-                leftParen: leftParen,
-                rightParen: rightParen,
-                as: asToken,
-                returnTypeExpression: typeExpression
-            });
 
             //support ending the function with `end sub` OR `end function`
-            func.body = this.block();
+            let body = this.block();
             //if the parser was unable to produce a block, make an empty one so the AST makes some sense...
-            if (!func.body) {
-                func.body = new Block({
-                    statements: [],
-                    startingRange: util.createRangeFromPositions(func.range.start, func.range.start)
-                });
-            }
-            func.body.symbolTable = new SymbolTable(`Block: Function '${name?.text ?? ''}'`, () => func.getSymbolTable());
-
-            if (!func.body) {
```

Comment:
This code was never called --- if `this.block()` failed, a new Block was created automatically.